framework/sandstone.cpp

```diff
@@ -1682,24 +1691,23 @@ static void wait_for_children(ChildrenList &children, int *tc, const struct test
         MonotonicTimePoint deadline = steady_clock::now() + remaining;
         for ( ; children_left && remaining > 0s; remaining = deadline - steady_clock::now()) {
             int ret = single_wait(ceil<milliseconds>(remaining));
-            if (ret >= 0)
-                continue;
-
-            for (ChildExitStatus &result : children.results)
-                result = { TestResult::Interrupted };
-
-            // Problem waiting: we must have caught a signal
-            // (child has likely not been able to write results)
-            int sig = ret;
-
-            logging_print_results(children.results, tc, test);
-            logging_printf(LOG_LEVEL_QUIET, "exit: interrupted\n");
-            logging_flush();
+            // handle "immediate" signals
+            if (ret > 0) {
+                logging_print_results(children.results, tc, test);
+                if (ret == SIGINT) {
+                    logging_printf(LOG_LEVEL_QUIET, "exit: interrupted\n");
+                } else if (ret == SIGTERM) {
+                    logging_printf(LOG_LEVEL_QUIET, "exit: terminated\n");
+                } else {
+                    logging_printf(LOG_LEVEL_QUIET, "exit: killed\n");
+                }
```

Comment:
This changes our reporting `exit` with two new values we don't currently have. Right now, "interrupted" means we stopped the testing before conclusion, regardless of which signal caused it -- it just happens to match the description for SIGINT.